qa/workunits/windows/py_tests/rbd_wnbd/service_restart_test.py

```diff
@@ -0,0 +1,235 @@
+                    help='The mechanism used to validate the written '
+                         'data. Examples: crc32c, md5, sha1, null, etc. '
+                         'If set to null, the written data will not be '
+                         'verified.',
+                    default='crc32c')
+parser.add_argument('--bs',
+                    help='Benchmark block size.',
+                    default="2M")
+parser.add_argument('--op',
+                    help='Benchmark operation. '
+                         'Examples: read, randwrite, rw, etc.',
+                    default="rw")
+parser.add_argument('--image-prefix',
+                    help='The image name prefix.',
+                    default="cephTest-")
+parser.add_argument('--image-size-mb',
+                    help='The image size in megabytes.',
+                    default=32, type=int)
+parser.add_argument('--map-timeout',
+                    help='Image map timeout.',
+                    default=60, type=int)
+parser.add_argument('--skip-enabling-disk', action='store_true',
+                    help='If set, the disk will not be turned online and the '
+                         'read-only flag will not be removed. Useful when '
+                         'the SAN policy is set to "onlineAll".')
+parser.add_argument('--verbose', action='store_true',
+                    help='Print info messages.')
+parser.add_argument('--debug', action='store_true',
+                    help='Print debug messages.')
+parser.add_argument('--stop-on-error', action='store_true',
+                    help='Stop testing when hitting errors.')
+parser.add_argument('--skip-cleanup-on-error', action='store_true',
+                    help='Skip cleanup when hitting errors.')
+
+
+class TestRunner(object):
+    def __init__(self,
+                 test_cls: typing.Type[stress_test.RbdTest],
+                 test_params: dict = {},
+                 iterations: int = 1,
+                 image_count: int = 8,
+                 workers: int = 1,
+                 stop_on_error: bool = False,
+                 cleanup_on_error: bool = True):
+        self.test_cls = test_cls
+        self.test_params = test_params
+        self.iterations = iterations
+        self.image_count = image_count
+        self.workers = workers
+        self.lock = threading.Lock()
+        self.completed = 0
+        self.errors = 0
+        self.stopped = False
+        self.stop_on_error = stop_on_error
+        self.cleanup_on_error = cleanup_on_error
+
+        self.test_instances: list[stress_test.RbdTest] = []
+
+    @Tracer.trace
+    def initialize(self):
+        LOG.info("Initializing mappings")
+
+        tg = task_group.TaskGroup(max_workers=self.workers)
+
+        for idx in range(self.image_count):
+            test = self.test_cls(**self.test_params)
+            self.test_instances.append(test)
+
+            tg.submit(test.initialize,
+                      stop_on_error=self.stop_on_error)
```

Comment:
Ouch, my bad. That was supposed to go in the TaskGroup constructor.

``cleanup_on_error`` does't have to be propagated to TaskGroup since we're handling it in the ``TestRunner.run`` method in the ``finally`` branch. Since we aren't actually using the optional cleanup hook in TaskGroup, I think I can go ahead and actually remove it for now.